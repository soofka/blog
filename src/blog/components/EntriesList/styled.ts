import styledComponents from 'styled-components';

export const StyledEntryListItem = styledComponents.div`
  margin: ${(props) => props.theme.marginM} 0;
  padding: ${(props) => props.theme.paddingM};
  box-sizing: border-box;

  &:first-of-type {
    margin-top: 0;
  }

  &:last-of-type {
    margin-bottom: 0;
  }
`;
