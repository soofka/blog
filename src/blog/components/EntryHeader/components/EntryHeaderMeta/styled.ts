import styledComponents from 'styled-components';

export const StyledEntryHeaderMeta = styledComponents.p`
  font-size: ${(props) => props.theme.fontSizeS};
  line-height: ${(props) => props.theme.lineHeightS};
  margin: ${
    (props) => `${props.theme.marginS} ${props.theme.marginXXS} ${props.theme.marginM} ${props.theme.marginXXS}`
  };
  padding: 0;
  color: ${(props) => props.theme.lighterBlack};

  span {
    font-weight: bold;
  }
`;

export default StyledEntryHeaderMeta;
