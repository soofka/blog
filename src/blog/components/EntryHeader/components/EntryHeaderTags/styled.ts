import styledComponents from 'styled-components';

export const StyledEntryHeaderTags = styledComponents.p`
  font-size: ${(props) => props.theme.fontSizeS};
  line-height: ${(props) => props.theme.lineHeightL};
  margin: 0 0 ${(props) => props.theme.marginM} 0;
  padding: 0;

  a {
    display: inline-block;
    margin: ${(props) => props.theme.marginXXS};
    padding: ${(props) => `${props.theme.paddingXS} ${props.theme.paddingS}`};
    background-color: ${(props) => props.theme.darkWhite};
    color: ${(props) => props.theme.blue};
    text-decoration: none;

    &:hover {
      background-color: ${(props) => props.theme.darkerWhite};
      color: ${(props) => props.theme.darkerBlue};
    }
  }
`;
