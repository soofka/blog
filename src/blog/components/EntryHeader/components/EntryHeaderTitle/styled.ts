import styledComponents from 'styled-components';

export const StyledEntryHeaderTitle = styledComponents.h1`
  font-size: ${(props) => props.theme.fontSizeXL};
  line-height: ${(props) => props.theme.lineHeightM};
  margin: ${(props) => props.theme.marginS} 0;
  padding: 0;

  a {
    color: ${(props) => props.theme.blue};
    text-decoration: none;

    &:hover {
      color: ${(props) => props.theme.lighterBlue};
    }
  }
`;

export default StyledEntryHeaderTitle;
