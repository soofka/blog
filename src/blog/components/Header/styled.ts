import styledComponents from 'styled-components';

export const StyledHeader = styledComponents.header`
  position: relative;
  display: inline-block;
  width: 100%;
  height: ${(props) => props.theme.headerHeight};
  color: ${(props) => props.theme.darkerWhite};
  background-color: ${(props) => props.theme.blue};
  box-shadow: ${(props) => `0 0 ${props.theme.baseDistanceValue}px 0 ${props.theme.darkerBlue}`};

  a {
    color: ${(props) => props.theme.darkerWhite};
    text-decoration: none;

    &:hover,
    &:active {
      color: ${(props) => props.theme.white};
    }
  }

  @media (max-width: ${(props) => props.theme.breakpointM}) {
    height: ${(props) => props.theme.headerHeightMobile};
  }
`;

export default StyledHeader;
