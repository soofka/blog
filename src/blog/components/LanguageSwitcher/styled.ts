import styledComponents from 'styled-components';

export const StyledLanguageSwitcher = styledComponents.ul`
  margin: ${(props) => props.theme.marginM};
  padding: 0;
  text-align: center;

  li {
    display: inline-block;
    margin: ${(props) => props.theme.marginS};

    svg {
      width: 48px;
      height: 27px;
      opacity: 0.5;
      cursor: pointer;

      .active {
        opacity: 0.75;
      }

      &:hover,
      &:active {
        opacity: 1;
      }
    }
  }

  @media (max-width: ${(props) => props.theme.breakpointM}) {
    float: right;
    margin: ${(props) => props.theme.marginS} 0;

    li {
      margin: ${(props) => `${props.theme.marginXS} ${props.theme.marginS}`};
      padding: 0;
    }
  }
`;
