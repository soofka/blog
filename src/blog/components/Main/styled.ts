import styledComponents from 'styled-components';

export const StyledMain = styledComponents.main`
  float: left;
  margin: 0 0 0 ${(props) => props.theme.infoSectionWidth};
  padding: ${(props) => props.theme.paddingM};
  width: 100%;
  height: auto;
  min-height: 100%;
  box-sizing: border-box;
  color: ${(props) => props.theme.lighterBlack};
  background-color: ${(props) => props.theme.white};
  box-shadow: ${(props) => `0 0 ${props.theme.baseDistanceValue}px 0 ${props.theme.white}`};

  a {
    color: ${(props) => props.theme.lighterBlack};

    &:hover,
    &:active {
      color: ${(props) => props.theme.black};
    }
  }

  small {
    display: block;
    width: 100%;
    font-size: ${(props) => props.theme.fontSizeS};
    line-height: ${(props) => props.theme.lineHeightS};
    margin: ${(props) => props.theme.marginM} 0;
    padding: 0;
    color: ${(props) => props.theme.gray};

    a {
      color: ${(props) => props.theme.gray};
      text-decoration: underline;

      &:hover,
      &:active {
        color: ${(props) => props.theme.gray};
        text-decoration: underline;
      }
    }
  }

  p,
  li {
    font-size: ${(props) => props.theme.fontSizeM};
    line-height: ${(props) => props.theme.lineHeightL};
    margin: 0 0 ${(props) => props.theme.marginL} 0;
    padding: 0;
    text-align: justify;

    a {
      color: ${(props) => props.theme.blue};

      &:hover,
      &:active {
        color: ${(props) => props.theme.lighterBlue};
      }
    }
  }

  li {
    margin: 0;
  }

  blockquote {
    color: ${(props) => props.theme.blue};
    margin: 0 ${(props) => `${props.theme.marginXL} ${props.theme.marginL} ${props.theme.marginXL}`};
    padding: 0 ${(props) => props.theme.paddingM};
    font-size: ${(props) => props.theme.fontSizeXXXL};
    line-height: ${(props) => props.theme.fontSizeXXXXL};
    border-left: 4px solid ${(props) => props.theme.lighterBlue};

    @media (max-width: ${(props) => props.theme.breakpointM}) {
      margin-left: 0;
      margin-right: 0;
    }
  }

  img {
    display: block;
    max-width: 100%;
    margin-left: auto;
    margin-right: auto;
    padding: 0;
  }

  ul {
    margin: 0;
    padding: 0 0 0 ${(props) => props.theme.paddingL};
  }

  pre code {
    background-color: ${(props) => props.theme.blue};
  }

  hr {
    width: 70%;
    height: 1px;
    margin: ${(props) => props.theme.marginXXL} auto;
    border: 0;
    border-top: 1px solid ${(props) => props.theme.darkerWhite};
  }

  .video-container {
    max-width: 800px;
    max-height: 450px;
    margin: ${(props) => `${props.theme.marginS} auto ${props.theme.marginM} auto`};
    overflow: hidden;
  }

  .video-container--wrapper {
    position: relative;
    padding-top: 24px;
    padding-bottom: 56.25%;
    height: 0;

    iframe {
      position: absolute;
      top: 0;
      left: 0;
      width: 100%;
      height: 100%;
    }
  }

@media (max-width: ${(props) => props.theme.breakpointM}) {
  margin: 0;
  padding: ${(props) => props.theme.paddingS};
  width: 100%;
}`;
