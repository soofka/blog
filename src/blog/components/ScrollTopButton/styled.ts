import styledComponents from 'styled-components';

export const StyledScrollTopButton = styledComponents.a`
  float: left;
  width: 100%;
  margin: ${(props) => props.theme.marginM} 0;
`;
