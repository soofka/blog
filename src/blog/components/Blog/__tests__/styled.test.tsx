import { shallow } from 'enzyme';
import 'jest-styled-components';
import * as React from 'react';

import { theme } from 'blog/common/theme';
import { StyledBlog } from '../styled';

describe('<StyledBlog/>', () => {

  it('is styled correctly', () => {
    expect(shallow(<StyledBlog theme={theme} />)).toMatchSnapshot();
  });

});
