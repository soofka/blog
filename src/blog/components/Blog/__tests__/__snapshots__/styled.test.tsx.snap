// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StyledBlog/> is styled correctly 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  font-size: 18px;
  line-height: 18px;
}

<div
  className="c0"
/>
`;
